third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -2153,35 +2153,36 @@ func flattenPrivateClusterConfig(c *containerBeta.PrivateClusterConfig) []map[st
 	}
 }
 
-func flattenIPAllocationPolicy(c *containerBeta.IPAllocationPolicy, d *schema.ResourceData, config *Config) []map[string]interface{} {
-	if c == nil {
+func flattenIPAllocationPolicy(c *containerBeta.Cluster, d *schema.ResourceData, config *Config) []map[string]interface{} {
+	if c == nil || c.IpAllocationPolicy == nil {
 		return nil
 	}
-    node_cidr_block := ""
-    if c.SubnetworkName != "" {
-		subnetwork, err := ParseSubnetworkFieldValue(c.SubnetworkName, d, config)
-        if err == nil {
-            sn, err := config.clientCompute.Subnetworks.Get(subnetwork.Project, subnetwork.Region, subnetwork.Name).Do()
-            if err == nil {
-                node_cidr_block = sn.IpCidrRange
-            }
-        } else {
+	nodeCidrBlock := ""
+	if c.Subnetwork != "" {
```

Comment:
Since every cluster has a subnetwork, won't we be reading this in cases where `create_subnetwork` isn't `true` as well? I think we need to guard on that as well.

`subnetwork_name` is only usable when `create_subnetwork` is `true`, so previously we were only filling in the value on a subset of cases where it was valid. Unless I'm missing something, after this change we'll fill it in whenever this block is returned.

Reply:
Why is that an issue if it's a computed field? (we still set the subnetwork name to the one returned in the ipallocationpolicy block, so it's only the nodeCidrBlock that gets filled in in more cases now)

Reply:
Oh! Ignore me, I didn't notice it was `Computed`.